### PR TITLE
Work to improve/expand destroy_pool testing a little

### PR DESCRIPTION
**Pull request by agrover**
`destroy_pool` should succeed if the pool has devs, but should fail if the pool has filesystems.

### CODE REVIEW

File: src/engine/engine.rs [hunk 1]
```diff
@@ -83,6 +83,8 @@ pub trait Pool: Debug {
     fn blockdevs(&mut self) -> Vec<&mut Dev>;
     fn cachedevs(&mut self) -> Vec<&mut Dev>;
 
+    fn destroy(self) -> EngineResult<()>;
```
Comment (mulkieran) [inline src/engine/engine.rs]: Needs a comment explaining its purpose.

File: src/engine/sim_engine/pool.rs [hunk 2]
```diff
@@ -81,6 +81,12 @@ impl Pool for SimPool {
         destroy_filesystems!{self; fs_names}
     }
 
+    fn destroy(mut self) -> EngineResult<()> {
+        self.block_devs.clear();
```
Comment (mulkieran) [inline src/engine/sim_engine/pool.rs]: Why is clearing these data structures necessary for SimPool and not for StratPool?

Comment (agrover) [inline src/engine/sim_engine/pool.rs]: true... should this function just do nothing?

Comment (agrover) [inline src/engine/sim_engine/pool.rs]: to answer my own q: yes, I believe it should do nothing 😃

File: src/engine/engine.rs (hunk 1 above)
Comment (agrover) [inline src/engine/engine.rs]: yup

File: src/engine/strat_engine/pool.rs [hunk 3]
```diff
@@ -157,6 +157,21 @@ impl Pool for StratPool {
         Ok(bdev_paths)
     }
 
+    fn destroy(self) -> EngineResult<()> {
+
+        // TODO: first, tear down DM mappings
+
+        for (_, mut bd) in self.block_devs {
```
Comment (mulkieran) [inline src/engine/strat_engine/pool.rs]: Minor: You could use ```values()``` like:

```
for mut bd in self.block_devs.values_mut() {
```
```

File: src/engine/engine.rs [hunk 4]
```diff
@@ -83,6 +83,11 @@ pub trait Pool: Debug {
     fn blockdevs(&mut self) -> Vec<&mut Dev>;
     fn cachedevs(&mut self) -> Vec<&mut Dev>;
 
+    /// Destroy the pool.
```
Comment (mulkieran) [inline src/engine/engine.rs]: What does it mean to destroy a pool?

File: src/engine/strat_engine/pool.rs (hunk 3 above)
Comment (agrover) [inline src/engine/strat_engine/pool.rs]: 👍

**Comment by mulkieran** [comment]
Now that ```wipe_header()``` is implemented in source for BlockDev it can be used in already implemented test code: https://github.com/stratis-storage/stratisd/blob/master/tests/util/blockdev_utils.rs#L35.

**Comment by mulkieran** [comment]
@agrover I'm still hoping for improvements to existing tests in test directroy by the use of newly introduced```BlockDev::wipe_header()``` where appropriate.

**Comment by agrover** [comment]
@mulkieran unfortunately it doesn't just drop in... `wipe_metadata` is a BlockDev method, but  in the tests `test_blockdev_setup` fn we are preparing the raw block devices -- we have a list of paths not a list of BlockDevs.

**Pull request merged**